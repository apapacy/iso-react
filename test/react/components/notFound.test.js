import React from 'react';
import { shallow, configure } from 'enzyme';
// import { expect } from 'chai';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import NotFound from '../../../src/react/components/notFound';

configure({ adapter: new Adapter() });

test('Link changes the class when hovered.', () => {
  const component = renderer.create(<NotFound />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  //tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  //tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('notFound has text "Not Found"', () => {
  const component = shallow(<NotFound />);
  expect(component.text()).toContain('Page not found');
});