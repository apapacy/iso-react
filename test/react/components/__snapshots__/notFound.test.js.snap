// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered. 1`] = `
<div>
  Page not found
</div>
`;

exports[`Link changes the class when hovered. 2`] = `
<div>
  Page not found
</div>
`;

exports[`Link changes the class when hovered. 3`] = `
<div>
  Page not found
</div>
`;
